frontend/src/core/entities/reducer.js

```diff
@@ -63,8 +63,26 @@ function injectSiblingEntities(
         entities[index],
         ...siblings.succeeding,
     ];
-    entities.splice(index, 1, ...list);
 
+    siblings.preceding.forEach((sibling) => {
+        const i = entities.findIndex((item) => item.pk === sibling.pk);
+        entities.forEach((e) => {
+            if (entities.indexOf(e) === i) {
+                list.splice(list.indexOf(sibling), 1);
+            }
+        });
+    });
+
+    siblings.succeeding.forEach((sibling) => {
+        const i = entities.findIndex((item) => item.pk === sibling.pk);
+        entities.forEach((e) => {
+            if (entities.indexOf(e) === i) {
+                list.splice(list.indexOf(sibling), 1);
+            }
+        });
+    });
+
+    entities.splice(index, 1, ...list);
```

Comment:
That should work, but we should optimize by reducing the number of loops and the lines of code. You could keep the original code, and just remove duplicates before returning.

See https://stackoverflow.com/a/56757215 for inspiration.

```js
entities = entities.filter((v, i, a) => a.findIndex(t => (t.id === v.id)) === i)
```

Reply:
Thanks! I didn't know you can add that many parameters in `filter()` learning something new every day 👍🏾